iib/web/errors.py

```diff
@@ -32,3 +34,38 @@ def json_error(error):
         response = jsonify({'error': msg})
         response.status_code = status_code
     return response
+
+
+def handle_broker_error(request):
+    """
+    Handle broker errors by setting the request as failed and raise an IIBError exception.
+
+    :param Request request: Request which will be set as failed
+    """
+    request.add_state("failed", "The scheduling of the request failed")
+    db.session.commit()
+    messaging.send_message_for_state_change(request)
+
+    error_message = f"The scheduling of the build request with ID {request.id} failed"
+    current_app.logger.exception(error_message)
+
+    raise IIBError(error_message)
+
+
+def handle_broker_batch_error(requests):
+    """
+    Handle broker errors by setting all request as failed and raise an IIBError exception.
+
+    :param list requests: list of all requests that should be marked as failed
```

Comment:
Could we please add the :raises IIBError... in the docstring ?

Reply:
Thanks, done.